File: julia/prime-factors/prime-factors.jl
function prime_factors(n)
    n == 1 && return []
    n == 2 && return [2]
    isprime(n) && return [n]

    result = []
    remaining = n
    
    while remaining % 2 == 0
        remaining /= 2
        push!(result, 2)
    end
    
    for i in 3:2:(n ÷ 2 + 1)
        while remaining % i == 0
            remaining /= i
            push!(result, i)
        end
        remaining == 1 && break
    end
    sort(result)
end

isprime(n) = !any(n % i == 0 for i in 2:(n ÷ 2 + 1))

File: r/prime-factors/prime-factors.R
prime_factors <- function(number) {
  result <- c()
  remaining <- number
  
  i <- 2
  while (i <= number && remaining > 1) {
    while (remaining %% i == 0) {
      result <- c(result, i)
      remaining <- remaining / i
      if (remaining == 1) break
    }
    i <- i + 1
  }
  result
}
